pkg/resolver/messages_test.go

```diff
@@ -0,0 +1,140 @@
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stolostron/search-v2-api/graph/model"
+)
+
+func Test_Messages_ValidCache(t *testing.T) {
+	// Build mock
+	mockMessage := Message{
+		cache: &MockCache{
+			disabled: map[string]struct{}{"managed1": {}},
+		},
+	}
+	//Execute the function
+	res, err := mockMessage.messageResults(context.Background())
+
+	// Validate
+	messages := make([]*model.Message, 0)
+	kind := "information"
+	desc := "Search is disabled on some of your managed clusters."
+	message := model.Message{ID: "S20",
+		Kind:        &kind,
+		Description: &desc}
+	messages = append(messages, &message)
+
+	if !reflect.DeepEqual(messages, res) {
+		t.Errorf("Message results doesn't match. Expected: %#v, Got: %#v", messages, res)
+	}
+	if err != nil {
+		t.Errorf("Incorrect results. expected error to be [%v] got [%v]", nil, err)
+	}
+}
+
+//no uid set in context - returns error
+func Test_Messages_Error(t *testing.T) {
+	// Build mock
+	mockMessage := Message{
+		cache: &MockCache{
+			disabled: map[string]struct{}{"managed1": {}},
+			err:      fmt.Errorf("err running query"),
+		},
+	}
+	//Execute the function
+	res, errRes := mockMessage.messageResults(context.Background()) //no uid set in context - returns error
+
+	// Validate
+	if !reflect.DeepEqual([]*model.Message{}, res) {
+		t.Errorf("Message results doesn't match. Expected: %#v, Got: %#v", []*model.Message{}, res)
+	}
+	if errRes == nil {
+		t.Errorf("Incorrect results. expected error to be [%v] got [%v]", fmt.Errorf("err running query"), errRes)
+	}
+}
+
+func Test_Message_Results_ValidCache(t *testing.T) {
+	// Build mock
+	mockMessage := Message{
+		cache: &MockCache{
+			disabled: map[string]struct{}{"managed1": {}},
+			err:      nil,
+		},
+	}
+	//Execute the function
+	res, err := mockMessage.messageResults(context.Background())
+
+	// Validate
+	messages := make([]*model.Message, 0)
+	kind := "information"
+	desc := "Search is disabled on some of your managed clusters."
+	message := model.Message{ID: "S20",
+		Kind:        &kind,
+		Description: &desc}
+	messages = append(messages, &message)
+
+	if !reflect.DeepEqual(messages, res) {
+		t.Errorf("Message results doesn't match. Expected: %#v, Got: %#v", messages, res)
+	}
+	if err != nil {
+		t.Errorf("Incorrect results. expected error to be [%v] got [%v]", nil, err)
+	}
+}
+
+// user does not have access to disabled clusters
+func Test_Message_Results_NoAccessToDisabledC(t *testing.T) {
+	// Build mock
+	mockMessage := Message{
+		cache: &MockCache{
+			disabled: map[string]struct{}{},
+			err:      nil,
+		},
+	}
+	//Execute the function
+	res, err := mockMessage.messageResults(context.Background())
+
+	// Validate
+	if !reflect.DeepEqual([]*model.Message{}, res) {
+		t.Errorf("Message results doesn't match. Expected: %#v, Got: %#v", []*model.Message{}, res)
+	}
+	if err != nil {
+		t.Errorf("Incorrect results. expected error to be [%v] got [%v]", nil, err)
+	}
+}
+
+//Return error running query
```

Comment:
Should this be removed?

Reply:
Kept it here in case I missed something, but this scenario should be covered in the rbac package tests.
Removing it.